fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -256,9 +269,14 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 	{
 		writeThread = createGenericThreadPool();
 		readThreads = createGenericThreadPool();
-		writeThread->addThread(new Writer(db, id));
+		if (g_network->isSimulated()) {
+			proc = g_simulator.getCurrentProcess();
+			mallopt(M_ARENA_MAX, 4);
+			nReaders = 1;
```

Comment:
The simulator was running out of virtual memory because glibc malloc uses a ton of virtual memory per thread.  I think I can either crank the numbers of readers back up or delete the mallopt call.

nReaders = 1 sounds more deterministic than having lots of readers, but setting the number of arenas to a low number causes mutex contention.  Any strong opinions?

Reply:
Having fewer readers in simulation is fine with me.